cupyx/scipy/ndimage/_filters_core.py

```diff
@@ -0,0 +1,368 @@
+import cupy
+
+
+def _get_output(output, input, shape=None):
+    if shape is None:
+        shape = input.shape
+    if isinstance(output, cupy.ndarray):
+        if output.shape != tuple(shape):
+            raise ValueError('output shape is not correct')
+    else:
+        dtype = output
+        if dtype is None:
+            dtype = input.dtype
+        output = cupy.zeros(shape, dtype)
+    return output
+
+
+def _fix_sequence_arg(arg, ndim, name, conv=lambda x: x):
+    if isinstance(arg, str):
+        return [conv(arg)] * ndim
+    try:
+        arg = iter(arg)
+    except TypeError:
+        return [conv(arg)] * ndim
+    lst = [conv(x) for x in arg]
+    if len(lst) != ndim:
+        msg = "{} must have length equal to input rank".format(name)
+        raise RuntimeError(msg)
+    return lst
+
+
+def _check_origin(origin, width):
+    origin = int(origin)
+    if (width // 2 + origin < 0) or (width // 2 + origin >= width):
+        raise ValueError('invalid origin')
+    return origin
+
+
+def _origins_to_offsets(origins, w_shape):
+    return tuple(x//2+o for x, o in zip(w_shape, origins))
+
+
+def _check_mode(mode):
+    if mode not in {'reflect', 'constant', 'nearest', 'mirror', 'wrap'}:
```

Comment:
I think the style preference is to use tuple; the uniqueness is already guaranteed:
```suggestion
    if mode not in ('reflect', 'constant', 'nearest', 'mirror', 'wrap'):
```